app/src/main/java/org/shadowice/flocke/andotp/Utilities/Settings.java

```diff
@@ -124,6 +125,10 @@ private long getLong(int keyId, long defaultValue) {
         return settings.getLong(getResString(keyId), defaultValue);
     }
 
+    private Set<String> getStringSet(int keyId, Set<String> defaultValue) {
+        return settings.getStringSet(getResString(keyId), defaultValue);
```

Comment:
You have to use the following code here:
```
return new HashSet<>(settings.getStringSet(getResString(keyId), defaultValue));
```
otherwise the modifying in `setTagToggle()` won't work. See [here](https://stackoverflow.com/questions/14034803/misbehavior-when-trying-to-store-a-string-set-using-sharedpreferences).